app/services/invoices/provider_taxes/pull_taxes_and_apply_service.rb

```diff
@@ -34,18 +34,19 @@ def call
         provider_taxes = taxes_result.fees
 
         ActiveRecord::Base.transaction do
+          unless invoice.draft?
+            invoice.issuing_date = issuing_date
+            invoice.payment_due_date = payment_due_date
```

Comment:
you are right, draft invoice can already have this informations. However, when we are in the finalize mode, we need to overwrite those informations since finalization can happen also before original issuing date (if it is triggered manually)